Marlin/src/HAL/AVR/HAL_SPI.cpp

```diff
@@ -35,22 +35,20 @@
 
 void spiBegin() {
   #if PIN_EXISTS(SD_SS)
-    OUT_WRITE(SD_SS_PIN, HIGH);
+    // Do not init HIGH for boards with pin 4 used as Fans or Heaters or otherwise, not likely to have multiple SPI devices anyway.
+    #if defined(__AVR_ATmega644__) || defined(__AVR_ATmega644P__) || defined(__AVR_ATmega644PA__) || defined(__AVR_ATmega1284P__)
+      // SS must be in output mode even it is not chip select
+      SET_OUTPUT(SD_SS_PIN);
+    #else
+      // set SS high - may be chip select for another SPI device
+      OUT_WRITE(SD_SS_PIN, HIGH);
+    #endif
   #endif
   SET_OUTPUT(SD_SCK_PIN);
   SET_INPUT(SD_MISO_PIN);
   SET_OUTPUT(SD_MOSI_PIN);
 
-  #if DISABLED(SOFTWARE_SPI)
-    // SS must be in output mode even it is not chip select
-    //SET_OUTPUT(SD_SS_PIN);
-    // set SS high - may be chip select for another SPI device
-    //#if SET_SPI_SS_HIGH
-      //WRITE(SD_SS_PIN, HIGH);
-    //#endif
-    // set a default rate
-    spiInit(1);
-  #endif
+  IF_DISABLED(SOFTWARE_SPI, spiInit(SPI_HALF_SPEED));
```

Comment:
I see that many instances of `spiBegin` call `spiInit` so maybe this isn't strictly needed.

Reply:
My apologies, I failed to sync the local to upstream on this PR.

I prefer we set it instead of hoping the latter will do that. This is, just a bug fix that deals with SET_SPI_SS_HIGH being ripped out previously. I don't want to introduce more surprises without going over it with a broader view.